internal/services/mssqlmanagedinstance/mssql_managed_database_data_source.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package mssqlmanagedinstance
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	// nolint: staticcheck
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssqlmanagedinstance/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/sql/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type MsSqlManagedDatabaseDataSourceModel struct {
+	Name                    string                    `tfschema:"name"`
+	ResourceGroupName       string                    `tfschema:"resource_group_name"`
+	ManagedInstanceName     string                    `tfschema:"managed_instance_name"`
+	ManagedInstanceId       string                    `tfschema:"managed_instance_id"`
+	LongTermRetentionPolicy []LongTermRetentionPolicy `tfschema:"long_term_retention_policy"`
+	ShortTermRetentionDays  int64                     `tfschema:"short_term_retention_days"`
+	PointInTimeRestore      []PointInTimeRestore      `tfschema:"point_in_time_restore"`
+}
+
+var _ sdk.DataSource = MsSqlManagedDatabaseDataSource{}
+
+type MsSqlManagedDatabaseDataSource struct{}
+
+func (d MsSqlManagedDatabaseDataSource) ResourceType() string {
+	return "azurerm_mssql_managed_database"
+}
+
+func (d MsSqlManagedDatabaseDataSource) ModelObject() interface{} {
+	return &MsSqlManagedDatabaseDataSourceModel{}
+}
+
+func (d MsSqlManagedDatabaseDataSource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.ManagedDatabaseID
+}
+
+func (d MsSqlManagedDatabaseDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ValidateMsSqlManagedInstanceDatabaseName,
+		},
+		"managed_instance_id": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ManagedInstanceID,
```

Comment:
We should use the validation function defined in the `commonids` module
```suggestion
			ValidateFunc: commonids.ValidateSqlManagedInstanceID,
```